### DISCUSSION TITLE
problem  resolving *.local names

### DISCUSSION THREAD

**Discussion by itakka01** [bluenviron/mediamtx#784] [Questions] [answered]
I am having a  problem when resolving *.local names: 

Cameras are on my local network get their names on the fly when connecting to the WIFI.
For example as soon as my camera with the local hostname  "raspicam-zero-a001" connects to the WIFI, it  can
can be seen on the LAN as  "raspicam-zero-a001.local"

ping raspicam-zero-a001.local
PING raspicam-zero-a001.local (192.168.1.5) 56(84) bytes of data.
64 bytes from raspicam-zero-a001 (192.168.1.5): icmp_seq=1 ttl=64 time=5.05 ms

But rtsp-simple-server does not accept these names. 

The following configuration leads to an error: 
---> cut out of: rtsp-simple-server.yml
[...]
paths:
  proxied:
    source: rtsp://raspicam-zero-a001.local:8554/unicast
[...]
---> cut

-> 2022/01/24 12:17:52 INF [path proxied] [rtsp source] ERR: dial tcp: lookup raspicam-zero-a001.local on 127.0.0.53:53: no such host

Put putting the real IP in the configuration is ok: 

---> cut out of: rtsp-simple-server.yml
[...]
paths:
  proxied:
    source: rtsp://192.168.1.5:8554/unicast
[...]
---> cut

What should I do that these .local names are also accepted?

regards
Bernhard

**Comment by aler9** [comment, marked as answer]
Hello, Go doesn't support mDNS natively, while [ping does](https://serverfault.com/questions/579967/why-nslookup-doesnt-use-mdns-while-ping-do). A solution consists in hacking the code and calling a [mDNS library](https://github.com/hashicorp/mdns) before every connection. This may be a good improvement for the server but we need to find out what's the behavior of other servers, whether they support mDNS or not.

**Comment by itakka01** [comment]
My idea was resolving the IP of the *.local cameras and then to feed  it with an exported variable to simple-rtsp-server.
But unfortunately (after reading the docs ;-) )  I could not find out how to do this.

Maybe someone can give me a hint how to use exported variables within the config file. 

---> cut  starting_bash_script
#!/bin/bash
hostname_rtsp_camera1="raspicam-zero-a001.local"
hostname_rtsp_camera2="raspicam-zero-a002.local"
hostname_rtsp_camera3="raspicam-zero-a003.local"
ip_rtsp_camera1=`ping -q -c 1 -t 1 ${hostname_rtsp_camera1} | grep PING | sed -e "s/).*//" | sed -e "s/.*(//"`
ip_rtsp_camera2=`ping -q -c 1 -t 1 ${hostname_rtsp_camera2} | grep PING | sed -e "s/).*//" | sed -e "s/.*(//"`
ip_rtsp_camera3=`ping -q -c 1 -t 1 ${hostname_rtsp_camera3} | grep PING | sed -e "s/).*//" | sed -e "s/.*(//"`
export RTSP_CAM1=${ip_rtsp_camera1}
export RTSP_CAM2=${ip_rtsp_camera2}
export RTSP_CAM3=${ip_rtsp_camera3}
./rtsp-simple-server ./rtsp-simple-server.yml
---> cut

---> cut out of: rtsp-simple-server.yml
[...]
paths:
  proxied_cam1:
    source: rtsp://$RTSP_CAM1:8554/unicast
  proxied_cam2:
    source: rtsp://$RTSP_CAM2:8554/unicast
  proxied_cam3:
    source: rtsp://$RTSP_CAM3:8554/unicast
 [...]
---> cut

**Reply by aler9** [reply]
you can do this with the `envsubst` Linux utility...

rtsp-simple-server-with-vars.yml:

```
paths:
proxied_cam1:
source: rtsp://$RTSP_CAM1:8554/unicast
proxied_cam2:
source: rtsp://$RTSP_CAM2:8554/unicast
proxied_cam3:
source: rtsp://$RTSP_CAM3:8554/unicast
```

command:

```
envsubst < rtsp-simple-server-with-vars.yml > rtsp-simple-server.yml
```

resulting rtsp-simple-server.yml:

```
paths:
proxied_cam1:
source: rtsp://cam1ip:8554/unicast
proxied_cam2:
source: rtsp://cam2ip:8554/unicast
proxied_cam3:
source: rtsp://cam3ip:8554/unicast
```

[a comment by an automated account is not shown]
